chains/evm/contracts/pools/FastTransferTokenPoolAbstract.sol

```diff
@@ -0,0 +1,431 @@
+
+import {CCIPReceiver} from "../applications/CCIPReceiver.sol";
+import {Client} from "../libraries/Client.sol";
+import {TokenPool} from "./TokenPool.sol";
+
+import {IERC20} from
+  "@chainlink/contracts/src/v0.8/vendor/openzeppelin-solidity/v4.8.3/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from
+  "@chainlink/contracts/src/v0.8/vendor/openzeppelin-solidity/v4.8.3/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC165} from
+  "@chainlink/contracts/src/v0.8/vendor/openzeppelin-solidity/v4.8.3/contracts/utils/introspection/IERC165.sol";
+
+/// @title Abstract Fast-Transfer Pool
+/// @notice Base contract for fast-transfer pools that provides common functionality
+/// for quoting, fill-tracking, and CCIP send helpers.
+abstract contract FastTransferTokenPoolAbstract is TokenPool, CCIPReceiver, ITypeAndVersion, IFastTransferPool {
+  using SafeERC20 for IERC20;
+
+  error WhitelistNotEnabled();
+  error InvalidLaneConfig();
+  error FillerNotWhitelisted(uint64 remoteChainSelector, address filler);
+
+  event LaneUpdated(
+    uint64 indexed dst,
+    uint16 bps,
+    uint256 fillAmountMaxPerRequest,
+    bytes destinationPool,
+    address[] addFillers,
+    address[] removeFillers
+  );
+  event FillerAllowListUpdated(uint64 indexed dst, address[] addFillers, address[] removeFillers);
+  event DestinationPoolUpdated(uint64 indexed dst, address destinationPool);
+
+  struct LaneConfig {
+    uint256 fillAmountMaxPerRequest; //    max amount that can be filled per request
+    uint16 bpsFastFee; // ─────────────╮ 0-10_000
+    bool fillerAllowlistEnabled; // ───╯ whitelist for fillers
+    bytes destinationPool; // destination pool address
+    mapping(address filler => bool isAllowed) fillerAllowList; // whitelist of fillers
+  }
+
+  struct LaneConfigView {
+    uint256 fillAmountMaxPerRequest; //    max amount that can be filled per request
+    uint16 bpsFastFee; // ────────────╮ 0-10_000
+    bool fillerAllowlistEnabled; // ──╯ whitelist for fillers
+    bytes destinationPool; // destination pool address
+  }
+
+  struct LaneConfigArgs {
+    uint256 fillAmountMaxPerRequest; //    max amount that can be filled per request
+    address[] addFillers; //               address allowed to fill
+    address[] removeFillers; //            addresses to remove from the whitelist
+    uint64 remoteChainSelector; // ──────╮
+    uint16 bpsFastFee; //                │ 0-10_000
+    bool fillerAllowlistEnabled; // ─────╯
+    bytes destinationPool;
+  }
+
+  struct MintMessage {
+    uint256 srcAmountToTransfer; // source amount from fill request
+    uint256 fastTransferFee; // fast transfer fee in the source token
+    bytes receiver; // receiver address on the destination chain
+    uint8 srcDecimals; // decimals of the source token
+  }
+
+  /// @notice Enum representing the state of a fill request
+  enum FillState {
+    NotFilled, // Request has not been filled yet
```

Comment:
nit: please capitalize so `NOT_FILLED`, `FILLED`, and `SETTLED` like Internal.MessageExecutionState